NuGetResources/NuGetPackageFeed3.cs

```diff
@@ -234,16 +234,25 @@ internal IEnumerable<PackageBase> Find(string registrationUrl, NuGetSearchContex
 
                 if (context.RequiredVersion == null && context.MinimumVersion == null && context.MaximumVersion == null && packages != null)
                 {
-                    PackageBase absoluteLatestPackage = packages.Where(p => p.IsPrerelease).OrderByDescending(pb => ((IPackage)pb).Version).FirstOrDefault();
-                    if (absoluteLatestPackage != null)
+                    // If all versions of a package are unlisted and a version flag is not specified, do not return any packages
+                    var allVersionsUnlisted = ((packages.Where(p => (!p.Published.HasValue || p.Published.Value.Year > 1900))).IsNullOrEmpty()) ? true : false;
```

Comment:
for future reference: in general, var a = expression ? true : false is redundant, because expression must be able to be converted to true/false for it to compile, so the ? true: false bit isn't needed. I would probably write


bool allVersionsUnlisted = packages.Where(...).IsNullOrEmpty();

But this is a style thing, not a big deal